Sorting/intersectSortedArrays.js

```diff
@@ -0,0 +1,44 @@
+//inputs sorted arrays, outputs an array of their intersection.
+//will duplicate repeats if they appear in the input arrays.
+//works a lot like mergeSort, using array pointers and changing the pointer value based on which condition is satisfied.
```

Comment:
very nice! 🚀  

Can you see the simple tweak to ensure there are no duplicates in the output, even if there are duplicates in the input?

Reply:
i need to add something like the following at the start of the while loop. do these nested while loops hurt my runtime too much? i haven't really studied the master theorem yet. i guess this is better than includes, which has to iterate through the whole array though.


    while (arr1[p1] === arr1[p1+1]) {p1++;}

    while (arr2[p2] === arr2[p2+1]) {p2++;}

am i on the right track?

Reply:
The master theorem is mainly used for analyzing runtime of recursive functions, so I don't think it's applicable here.

It is possible to do with no extra loops. You should be able to do it by keeping track of one more piece of information than what you have now.